src/components/Booklist.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+const Booklist = ({ books }) => {
+  const [bookList, setBookList] = useState(books);
+
+  useEffect(() => {
+    setBookList(books);
+  }, [books]);
+
+  const handleDeleteBook = (bookId) => {
+    // Filter out the book with the given ID
+    const updatedBooks = bookList.filter((book) => book.id !== bookId);
+    setBookList(updatedBooks);
+  };
+
+  return (
+    <div>
+      <h2>Book List</h2>
+      <ul>
+        {bookList.map((book) => (
+          <li key={book.id}>
+            <span>
+              <strong>Title:</strong>
+              {book.title}
+              ,
+              <strong>Author:</strong>
+              {book.author}
+              ,
+              <strong>Category:</strong>
+              {book.category}
+            </span>
+            <button type="button" onClick={() => handleDeleteBook(book.id)}>Delete</button>
+          </li>
```

Comment:
- [ ] It will be great to have a separate component called `book` that takes the book data as props and it is the one responsible for displaying the book details. So that whole part of the code that i have highlighted should be in a separate component called book.